control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -103,7 +103,14 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 		params.Image = hcp.Spec.Configuration.Image
 		params.Scheduler = hcp.Spec.Configuration.Scheduler
 	}
-	params.AdvertiseAddress = util.AdvertiseAddressWithDefault(hcp, config.DefaultAdvertiseAddress)
+
+	defaultAddresses := &util.DefaultAdvIps{
```

Comment:
It seems overkill to have a data structure for defaults and to pass them to the SetAdvertiseAddress function.
They can simply be separate function parameters.